addons/binding/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/handler/HeosPlayerHandler.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.heos.handler;
+
+import static org.openhab.binding.heos.internal.resources.HeosConstants.PID;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.heos.internal.api.HeosFacade;
+import org.openhab.binding.heos.internal.api.HeosSystem;
+import org.openhab.binding.heos.internal.resources.HeosPlayer;
+
+/**
+ * The {@link HeosPlayerHandler} handles the actions for a HEOS player.
+ * Channel commands are received and send to the dedicated channels
+ *
+ * @author Johannes Einig - Initial contribution
+ */
+public class HeosPlayerHandler extends HeosThingBaseHandler {
+
+    private String pid;
+    private HeosPlayer player = new HeosPlayer();
```

Comment:
Consider making both these field `final`

Reply:
From my side only the pid is final. The `HeosPlayer()` field is used to transport new information during initialization.

Reply:
Indeed, I missed that